transform-plugins/src/test/java/co/cask/hydrator/plugin/XMLParserTest.java

```diff
@@ -0,0 +1,315 @@
+
+    DataSetManager<Table> outputManager = getDataset(sinkTable);
+    List<StructuredRecord> outputRecords = MockSink.readOutput(outputManager);
+    Assert.assertEquals("OutputRecords", 3, outputRecords.size());
+    for (StructuredRecord record : outputRecords) {
+      if (record.get("category").equals("cooking")) {
+        Assert.assertEquals("Everyday Italian", record.get("title"));
+        Assert.assertEquals(null, record.get("price"));
+        Assert.assertEquals(null, record.get("year"));
+        Assert.assertEquals("<subcategory><type>Continental</type></subcategory>", record.get("subcategory"));
+      } else if (record.get("category").equals("children")) {
+        Assert.assertEquals("Harry Potter", record.get("title"));
+        Assert.assertEquals(49.99, (Double) record.get("price"), 0.0);
+        Assert.assertEquals(2005, record.get("year"));
+        Assert.assertEquals("<subcategory><type>Series</type></subcategory>", record.get("subcategory"));
+      } else if (record.get("category").equals("web")) {
+        Assert.assertEquals("Learning XML", record.get("title"));
+        Assert.assertEquals(39.95, (Double) record.get("price"), 0.01);
+        Assert.assertEquals(2003, record.get("year"));
+        Assert.assertEquals("<subcategory><type>Basics</type></subcategory>", record.get("subcategory"));
+      }
+    }
+  }
+
+  @Test
+  public void testEmitErrors() throws Exception {
+    XMLParser.Config config = new XMLParser.Config("body", "UTF-8", "title:/book/title,author:/book/author," +
+      "year:/book/year", "title:string,author:int,year:int", "Write to error dataset");
+    Transform<StructuredRecord, StructuredRecord> transform = new XMLParser(config);
+    transform.initialize(new MockTransformContext());
+
+    StructuredRecord inputRecord = StructuredRecord.builder(INPUT)
+      .set("offset", 1)
+      .set("body", "<book category=\"COOKING\"><title lang=\"en\">Everyday Italian</title>" +
+        "<author>Giada De Laurentiis</author><year>2005</year><price>30.00</price></book>").build();
+
+    MockEmitter<StructuredRecord> emitter = new MockEmitter<>();
+    transform.transform(inputRecord, emitter);
+    Assert.assertEquals(0, emitter.getEmitted().size());
+    Assert.assertEquals(1, emitter.getErrors().size());
+
+    InvalidEntry<StructuredRecord> invalidEntry = emitter.getErrors().get(0);
+    Assert.assertEquals(31, invalidEntry.getErrorCode());
+    Assert.assertEquals("offset", 1, invalidEntry.getInvalidRecord().get("offset"));
+    Assert.assertEquals("body", "<book category=\"COOKING\"><title lang=\"en\">Everyday Italian</title><author>Giada " +
+      "De Laurentiis</author><year>2005</year><price>30.00</price>" +
+      "</book>", invalidEntry.getInvalidRecord().get("body"));
+  }
+
+  @Test
+  public void testIgnoreError() throws Exception {
+    XMLParser.Config config = new XMLParser.Config("body", "UTF-8", "title:/book/title,author:/book/author," +
```

Comment:
Can we use a different encoding (other than UTF-8 - say UTF-16) for one of the test cases? Want to make sure the other encoding values work.